### PR TITLE
dev/mail#32 - Further checks - making mailing test email non-case-sensitive

### PR DESCRIPTION
**Pull request by martinh-pw**
As per https://lab.civicrm.org/dev/mail/issues/32
Further work in addition to https://github.com/civicrm/civicrm-core/pull/13392

No longer assuming existing contact emails will be lowercase in the db, now will match them in a case insensitive way as well.

### CODE REVIEW

File: api/v3/Mailing.php [hunk 1]
```diff
@@ -637,7 +637,7 @@ function civicrm_api3_mailing_send_test($params) {
     $query = CRM_Utils_SQL_Select::from('civicrm_email e')
         ->select(array('e.id', 'e.contact_id', 'e.email'))
         ->join('c', 'INNER JOIN civicrm_contact c ON e.contact_id = c.id')
-        ->where('e.email IN (@emails)', array('@emails' => $testEmailParams['emails']))
+        ->where('LOWER(e.email) IN (@emails)', array('@emails' => $testEmailParams['emails']))
```
Comment (eileenmcnaughton) [inline api/v3/Mailing.php]: actually we don't need this LOWER - mysql is case insensitive - it's php we need to wrangle.

Note that using LOWER will significantly slow down the query - a slightly similar example is in this blog https://civicrm.org/blog/eileen/guppies-have-died-for-that-query (diff function but same underlying issue)

**Comment by jusfreeman** [comment]
@martinh-pw thanks for this fix

@eileenmcnaughton I've tested this PR with your amendment to remove LOWER from SQL. Can confirm this work. Nice fix.

**Comment by martinh-pw** [comment]
@eileenmcnaughton @jusfreeman Thanks :)

Sorry but I'm unfamiliar with the process here, is there a way I should modify this PR somehow? Or to close it somehow and create a new one?

**Comment by eileenmcnaughton** [comment]
@martinh-pw yes you can modify this - make the change & then do 

``` git add -p```
``` git commit --amend```

Then you will have fixed up the commit.

You can then force push it to your branch with

```
git push -f myrepo mybranch
```

**Comment by martinh-pw** [comment]
@eileenmcnaughton Done, nice trick!

**Comment by eileenmcnaughton** [comment]
looks good

**Pull request merged**